### ISSUE TITLE
If similarly named functions are found in >1 namespaces supplied by `force`, the namespaces are concatenated in the output

### ISSUE THREAD

**Issue by yogat3ch** [yonicd/sinew#62]
Reprex:

```
pretty_namespace(text = "rename(a, b = 'a')", force = list(dplyr = "rename", pryr = "rename"))
```
>dplyr::renampryr::rename(a, b = 'a')

I'm currently working on creating a prompt when there's a namespace overlap in `force` just like the one when there's an overlap in the results from `utils::help.search`

**Comment by yonicd** [comment]
thanks

there is a test branch that is being worked on to merge into the master. I'll merge what is in there so you can add tests if you want.

**Comment by yogat3ch** [comment]
Ok! Yeah, the PRs will definitely need `tests` to keep the functionality in check. I see about making those once I figure out how to resolve this bug. I think I might have messed up a number of documents with this bug since I've been using `sinew` on all of the R docs I've created over the past couple years. 
I think the new find & replace in files feature in Rstudio is going to come in handy for fixing it.

**Comment by yogat3ch** [comment]
I think [pretty_utils.R#100](https://github.com/yonicd/sinew/blob/46e01248dcdc6cb186bc03785b7a6350a86c4295/R/pretty_utils.R#L100) could be to blame for this issue...

**Comment by yonicd** [comment]
I think a simple solution would to not let the user enter conflicting namespaces.

options to solve would be to use one and warn or force a break and tell user to fix.

**Comment by yonicd** [comment]
validation function added https://github.com/yonicd/sinew/commit/45c1d074f435e2f870c884021e9297406470964c

**Comment by yogat3ch** [comment]
That works! 
I was working on something that was parsing `sym.funs` and giving a choice if `ask = TRUE` and there was a conflict in namespaces. Otherwise just warning and choosing the first if `ask = FALSE`.

**Comment by yogat3ch** [comment]
If you want to just give a `warning` and remove the duplicated objects in force, here's a function that I just used for my own purposes to remove them:

```
remove_duplicates <- function (l) {
  .l <- length(l)
  e <- environment()
  if(anyDuplicated(unlist(l))) warning("Duplicates found in force. Removing.", immediate. = TRUE, call. = FALSE)
  
  for (.li in 1:(.l - 1)) {
    .is <- ifelse(.li < .l, .li + 1, .li) : .l
    mapply(l[.is], names(l[.is]), FUN = function(.x, .y) {
      .keep <- setdiff(.x, l[[.li]])
      
      if (length(.keep) > 0) {
        if (length(.keep) != length(.x)) message(paste0("Removing ", paste0(.x[!.x %in% .keep], collapse = ", "), " from ", .y))
        e$l[[.y]] <<- .keep
        } else 
          NULL
    })
  }
  return(e$l)
}
```

**Comment by yonicd** [comment]
Thanks. I think being strict on the input is prudent here. The user should have some self control on using the functionality as intended

**Comment by yogat3ch** [comment]
I think the use case is when doing multiple documents or a directory of documents. 
I think that most users tend to have a good idea of the packages they prefer, and in what order they typically resort to them.
Like personally, I'm using a lot of `dplyr` so it's functions should take precedence over those in `stats`. 
I use a lot of functions from `dplyr`, so remembering and writing them all out to construct the `force` argument would be very time consuming.

So based on what I do know of which packages I prefer, I constructed a list that just used all od the exports of each of my go-to packages in order of priority (I think I used the function `getNamespaceExports`.
I didn't realize there were duplicates until I caught that double pasting error.

It makes me think about the usage of `force` more generally, because I can't imagine I'm the only one who remembers the packages moreso than I remember individual functions. Maybe `force` might match user experience more closely if there was the option to supply package names in order of precedence? Like if the user specified a `TRUE` for the list item named according to the package, it would just use all the exports from that package?

Just a thought. Im tempted to look through the closed issues to see if there's any indication of how people are using the package and if using it on an archive of documents is a common use case or not.
